src/Native/Unix/System.Native/pal_io.c

```diff
@@ -49,7 +49,7 @@ extern int     getpeereid(int, uid_t *__restrict__, gid_t *__restrict__);
 extern ssize_t  getline(char **, size_t *, FILE *);
 #endif
 
-#if HAVE_STAT64
+#if HAVE_STAT64 && !defined(__DARWIN_ONLY_64_BIT_INO_T)
```

Comment:
Can you explain this change?  I'm wondering if it wouldn't be better to update the cmake configuration so that HAVE_STAT64 is only set when STAT64 can actually be used.

Reply:
The `__DARWIN_ONLY_64_BIT_INO_T` define is explained in the `stat64` man page, see for instance https://www.unix.com/man-page/osx/2/stat64/.

On iOS, the `stat64()` function exists but `struct stat64` does not - and `stat()` is either 32-bit or 64-bit depending on device / simulator you're running on.

We could probably add a cmake check for `struct stat64`.  At the moment we are using this in Mono:

        AC_CHECK_FUNC(stat64,            [AC_DEFINE(HAVE_STAT64, 1, [stat64])])

Reply:
> The `__DARWIN_ONLY_64_BIT_INO_T` define is explained in the stat64 man page, see for instance https://www.unix.com/man-page/osx/2/stat64/.

I do not see this define explained in this page. It talks about other similar defines, but not this one.